src/game.cpp

```diff
@@ -856,45 +852,51 @@ bool game::start_game()
     return true;
 }
 
-bool game::place_starting_vehicle()
+static bool compare_distance( const tripoint &goal_a, const tripoint &goal_b )
+{
+    tripoint origin = g->u.global_omt_location();
+    return rl_dist( origin, goal_a ) < rl_dist( origin, goal_b );
+}
+
+void game::place_starting_vehicle( cata::optional<vproto_id> starting_vehicle )
 {
-    int count = 1;
-    while( count < 11 ) {
+    if( !starting_vehicle ) {
+        return;
+    }
+    omt_find_params find_params;
+    find_params.must_see = false;
+    find_params.cant_see = false;
+    find_params.types.emplace_back( "road", ot_match_type::type );
+    for( int count = 0; count < 10; ++count ) {
         // find nearest road
-        omt_find_params find_params;
+        find_params.min_distance = count * 2;
         find_params.search_range = 10 + count * 2;
-        find_params.min_distance = 0 + count * 2;
-        find_params.must_see = false;
-        find_params.cant_see = false;
-        std::vector<std::pair<std::string, ot_match_type>> types;
-        std::pair<std::string, ot_match_type> type_pair;
-        type_pair.first = "road";
-        type_pair.second = ot_match_type::type;
-        types.push_back( type_pair );
-        find_params.types = types;
-        find_params.existing_only = false;
         // if player spawns underground, park their car on the surface.
         tripoint omt_origin = u.global_omt_location();
         omt_origin.z = 0;
-        const tripoint goal = overmap_buffer.find_closest( omt_origin, find_params );
-        if( goal != overmap::invalid_tripoint ) {
-            // try place vehicle there.
-            tinymap target_map;
-            target_map.load( omt_to_sm_copy( goal ), false );
-            const tripoint &origin = target_map.getlocal( sm_to_ms_copy( omt_to_sm_copy( goal ) ) ) + point(
-                                         SEEX, SEEY );
-            vehicle *veh = target_map.add_vehicle( *u.starting_vehicle, origin, rng( -90, 90 ), rng( 50, 80 ),
-                                                   0,
-                                                   false );
-            if( veh ) {
-                overmap_buffer.add_vehicle( veh, target_map.getabs( origin ).xy() );
-                target_map.save();
-                return true;
+        std::vector<tripoint> goals = overmap_buffer.find_all( omt_origin, find_params );
+        std::sort( goals.begin(), goals.end(), compare_distance );
+        for( const tripoint &goal : goals ) {
+            if( goal != overmap::invalid_tripoint ) {
+                // try place vehicle there.
+                tinymap target_map;
+                target_map.load( omt_to_sm_copy( goal ), false );
+                const tripoint &origin = target_map.getlocal( sm_to_ms_copy( omt_to_sm_copy( goal ) ) ) + point(
+                                             SEEX, SEEY );
+                vehicle *veh = target_map.add_vehicle( *u.starting_vehicle, origin, rng( -90, 90 ), rng( 50, 80 ),
```

Comment:
Why is this reading a property of `u`?  What's the point of the parameter `starting_vehicle`?
```suggestion
                vehicle *veh = target_map.add_vehicle( *starting_vehicle, origin, rng( -90, 90 ), rng( 50, 80 ),
```